libs/dev-dock/backend/src/backend.ts

```diff
@@ -0,0 +1,176 @@
+}
+async function clearUsbDrive(): Promise<void> {
+  await execFile('sudo', [
+    join(MOCK_USB_SCRIPT_DIRECTORY, 'initialize.sh'),
+    '-s',
+    '100',
+  ]);
+}
+
+const DEV_DOCK_FILE_PATH = '/tmp/dev-dock.json';
+interface DevDockFileContents {
+  electionInfo?: DevDockElectionInfo;
+}
+
+function writeDevDockFileContents(fileContents: DevDockFileContents): void {
+  return fs.writeFileSync(DEV_DOCK_FILE_PATH, JSON.stringify(fileContents));
+}
+
+function readDevDockFileContents(): DevDockFileContents {
+  return JSON.parse(
+    fs.readFileSync(DEV_DOCK_FILE_PATH, { encoding: 'utf-8' })
+  ) as DevDockFileContents;
+}
+
+const api = grout.createApi({
+  setElection(input: { path?: string }): void {
+    if (!input.path) {
+      return writeDevDockFileContents({
+        ...readDevDockFileContents(),
+        electionInfo: undefined,
+      });
+    }
+
+    const electionData = fs.readFileSync(
+      electionPathToAbsolute(input.path),
+      'utf-8'
+    );
+    const parseResult = safeParseElectionDefinition(electionData);
+    assert(parseResult.isOk());
+    const electionInfo: DevDockElectionInfo = {
+      path: input.path,
+      title: parseResult.ok().election.title,
+    };
+
+    writeDevDockFileContents({
+      electionInfo,
+    });
+  },
+
+  getElection(): Optional<DevDockElectionInfo> {
+    return readDevDockFileContents().electionInfo;
+  },
+
+  getCardStatus(): CardStatus {
+    return readFromCardMockFile().cardStatus;
+  },
+
+  async insertCard(input: { role: DevDockUserRole }): Promise<void> {
+    const electionDefinitionPath = readDevDockFileContents().electionInfo?.path;
+    if (!electionDefinitionPath) {
+      return;
+    }
+
+    await execFile(MOCK_CARD_SCRIPT_PATH, [
+      '--card-type',
+      input.role.replace('_', '-'),
+      '--electionDefinition',
+      electionPathToAbsolute(electionDefinitionPath),
+    ]);
+  },
+
+  async removeCard(): Promise<void> {
+    await execFile(MOCK_CARD_SCRIPT_PATH, ['--card-type', 'no-card']);
+  },
+
+  getUsbDriveStatus(): DevDockUsbDriveStatus {
+    return getUsbDriveStatus();
+  },
+
+  async insertUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'removed') {
+      await insertUsbDrive();
+    }
+  },
+
+  async removeUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'inserted') {
+      await removeUsbDrive();
+    }
+  },
+
+  async clearUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'inserted') {
+      await removeUsbDrive();
+      await clearUsbDrive();
+      await insertUsbDrive();
+    } else {
+      await clearUsbDrive();
+    }
+  },
+});
+
+/**
+ * A type to be used by the frontend to create a Grout API client
+ */
+export type Api = typeof api;
+
+/**
+ * Mounts the Dev Dock API endpoints at /dock.
+ */
+export function useDevDockRouter(
+  app: Express.Application,
+  express: typeof Express
+): void {
+  if (!isFeatureFlagEnabled(BooleanEnvironmentVariableName.ENABLE_DEV_DOCK)) {
+    return;
+  }
```

Comment:
Maybe we could throw an error if `ENABLE_DEV_DOCK` is on but `MOCK_SMART_CARDS` is not?

Reply:
It might be useful to allow the other dev dock tools even while using real smartcards. I'd rather just disable the card interface